src/collector/rrdp.rs

```diff
@@ -510,369 +487,272 @@ impl<'a> Run<'a> {
-    fn tree_name(rpki_notify: &uri::Https) -> Vec<u8> {
-        format!("rrdp:{}", rpki_notify).into()
-    }
-
-    /// Returns the rpkiNotify for a given RRDP tree name.
-    ///
-    /// If the tree name isn’t actually for an RRDP collector tree, returns
-    /// `None`.
-    fn tree_uri(tree_name: &[u8]) -> Option<uri::Https> {
-        if tree_name.starts_with(b"rrdp:") {
-            uri::Https::from_slice(&tree_name[5..]).ok()
-        }
-        else {
-            None
-        }
-    }
-
     /// Loads an object from the repository.
     ///
     /// The object is identified by its rsync URI. If the object doesn’t
     /// exist, returns `None`.
-    pub fn load_file(
+    pub fn load_object(
         &self,
         uri: &uri::Rsync
     ) -> Result<Option<Bytes>, Failed> {
-        match self.tree.get(uri.as_str())? {
-            Some(value) => {
-                RepositoryObject::try_from(value).map(|obj| {
-                    Some(obj.content)
-                }).map_err(|_| {
-                    error!("Encountered invalid object in RRDP database.");
-                    Failed
-                })
-            }
-            None => Ok(None)
-        }
+        RepositoryObject::load(&self.object_path(uri)).map(|obj| {
+            obj.map(|obj| obj.content)
+        })
     }
 
-    /// Iterators over all the objects in the repository.
-    ///
-    /// Warns about broken objects but continues.
-    pub fn iter_files(
-        &self
-    ) -> impl Iterator<Item = (uri::Rsync, Bytes)> {
-        let mut failed = false;
-        self.tree.iter().map(move |item| {
-            let (key, value) = match item {
-                Ok(item) => item,
-                Err(err) => {
-                    if !failed { 
-                        warn!("Database error: {}", err);
-                        failed = true;
-                    }
-                    return None
-                }
-            };
-            if key == REPOSITORY_STATE_KEY {
-                return None
-            }
-            let uri = match uri::Rsync::from_slice(&key) {
-                Ok(uri) => uri,
-                Err(_) => {
-                    warn!(
-                        "Object with bad URI '{}' in RRDP collector.",
-                        String::from_utf8_lossy(&key)
-                    );
-                    return None
-                }
-            };
-            let object = match RepositoryObject::try_from(value) {
-                Ok(object) => object,
-                Err(_) => {
-                    warn!(
-                        "Broken object for URI {} in RRDP collector.",
-                        uri
-                    );
-                    return None
-                }
-            };
-            Some((uri, object.content))
-        }).flatten()
+    /// Returns the path where all the objects live.
+    fn object_base(&self) -> PathBuf {
+        self.path.join("rsync")
```

Comment:
This corresponds to the `rsync:` scheme of the URI. Having this extra directory makes it possible to keep the temporary directory for snapshots and deltas under the repository directory. Which in turns makes it less likely that the directory move at the end of a snapshot updates fails because source and target are on different file systems.